WordPress/gutenberg
# Added header toolbar slotfill

## What?
Split from [16988](https://github.com/WordPress/gutenberg/issues/16988)

## How?
This PR introduces a new SlotFill location at the end of the toolbar.

## Testing Instructions

1. Checkout this branch and build
2. Add the sample code below in your theme or plugin
3. Confirm that a new icon appears and is clickable as seen in the screenshot

```
import { registerPlugin } from '@wordpress/plugins';
import { PluginHeaderToolbar } from '@wordpress/editor';

const MyHeaderToolbarPlugin = () => (
	<PluginHeaderToolbar
		className="plugin-header-toolbar-button"
		classContentName="plugin-header-toolbar-content"
		renderContent={ () => <div>Rendered content</div> }
	/>
);

registerPlugin( 'my-header-toolbar-plugin', {
	render: MyHeaderToolbarPlugin,
	icon: 'smiley',
} );

```
## Screenshots or screencast 
<img width="1019" alt="header-toolbar-slot" src="https://github.com/user-attachments/assets/184e8c45-ee93-4d7d-a55f-855b4c480bf7">
